examples/nodejs/get-set-batch-perf-test/perf-test.ts

```diff
@@ -0,0 +1,251 @@
+import {
+  CacheClient,
+  CacheFlush,
+  CacheSet,
+  DefaultMomentoLoggerFactory,
+  DefaultMomentoLoggerLevel,
+  MomentoLogger,
+  MomentoLoggerFactory,
+} from '@gomomento/sdk';
+import {createCache, getCacheClient} from './utils/cache';
+import {GetSetConfig, PerfTestConfiguration, PerfTestOptions, RequestType} from './utils/perf-test-options';
+import {calculateSummary, getElapsedMillis} from './utils/perf-test-utils';
+
+class PerfTest {
+  private readonly momentoClientPromise: Promise<CacheClient>;
+  private momento: CacheClient;
+  private readonly loggerFactory: MomentoLoggerFactory;
+  private readonly logger: MomentoLogger;
+  private readonly cacheItemTtlSeconds = 600;
+  private readonly options: PerfTestOptions;
+  private readonly testConfiguration: PerfTestConfiguration;
+  private readonly cacheName: string = 'js-perf-test';
+  private totalRequests = 0;
+  private totalItemSizeBytes = 0;
+  private completionTimes: number[] = [];
+
+  constructor(options: PerfTestOptions, testConfiguration: PerfTestConfiguration) {
+    this.loggerFactory = options.loggerFactory;
+    this.logger = this.loggerFactory.getLogger('get-set-batch-perf-test');
+    this.options = options;
+    this.testConfiguration = testConfiguration;
+    this.momentoClientPromise = this.createMomentoClient(options);
+  }
+
+  private createMomentoClient(options: PerfTestOptions): Promise<CacheClient> {
+    return getCacheClient(options.loggerFactory, options.requestTimeoutMs, this.cacheItemTtlSeconds);
+  }
+  async createCache(): Promise<void> {
+    this.momento = await this.momentoClientPromise;
+    await createCache(this.momento, this.cacheName, this.logger);
+  }
+
+  async flushCache(): Promise<void> {
+    const res = await this.momento.flushCache(this.cacheName);
+    if (res instanceof CacheFlush.Success) {
+      this.logger.info('Cache flushed successfully');
+    } else if (res instanceof CacheFlush.Error) {
+      throw res.innerException();
+    }
+  }
+
+  async runAsyncSetRequests(): Promise<void> {
+    for (const setConfig of this.testConfiguration.sets) {
+      const startTime = process.hrtime();
+      while (getElapsedMillis(startTime) < this.testConfiguration.minimumRunDurationSecondsForTests * 1000) {
+        await this.sendAsyncSetRequests(this.momento, setConfig);
+      }
+
+      calculateSummary(
+        this.totalRequests,
+        this.totalItemSizeBytes,
+        this.testConfiguration.minimumRunDurationSecondsForTests,
+        setConfig.batchSize,
+        setConfig.itemSizeBytes,
+        this.completionTimes,
+        RequestType.ASYNC_SETS,
+        this.logger
+      );
+      this.totalRequests = 0;
+      this.totalItemSizeBytes = 0;
+      this.completionTimes = [];
+    }
+  }
+
+  async runAsyncGetRequests(): Promise<void> {
+    for (const getConfig of this.testConfiguration.gets) {
+      const startTime = process.hrtime();
```

Comment:
I think you should have some code right here that ensures that there are values set for all the keys that this config is going to need before you start running the gets.